### PR TITLE
PLANNER-2180 ScoreCalculators become public API

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

File: download/upgradeRecipe/upgradeRecipe6.1.adoc [hunk 1]
```diff
@@ -940,7 +940,7 @@ public class CloudBalancingEasyScoreCalculator implements SimpleScoreCalculator<
 After in `*.java`:
 [source, java]
 ----
-import org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator;
+import org.optaplanner.core.api.score.calculator.EasyScoreCalculator;
```
Comment (triceo) [inline download/upgradeRecipe/upgradeRecipe6.1.adoc:943]:
```suggestion
import org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator;
```

File: download/upgradeRecipe/upgradeRecipe8.adoc [hunk 2]
```diff
@@ -121,8 +62,6 @@ In OptaPlanner 8, we have removed most uses of the `Serializable` marker interfa
 https://docs.optaplanner.org/latest/optaplanner-docs/html_single/index.html#integrationWithPersistentStorage[serialize
 via JSON or other formats].
 
-== OptaPlanner Core
```
Comment (rsynek) [inline download/upgradeRecipe/upgradeRecipe8.adoc:124]: Why removing this (and the next one) header? The idea is to help users navigate through the already long (and growing) document. Not everybody is interested in changes made to Benchmark, for instance.

Comment (triceo) [reply download/upgradeRecipe/upgradeRecipe8.adoc:124]: I spent 20 minutes on this. There is no way how to make the header structure logical while maintaining this.
I mean - I could introduce this second level and turn all the individual changes to a third level.
But then, from 8.0.0 to 8.1.0, there would be no third level anymore, the individual changes all on the second level, and the consistency would be gone.

Comment (triceo) [reply download/upgradeRecipe/upgradeRecipe8.adoc:124]: Furthermore, as far as I can see, 7.x upgrade recipe didn't do this either.

**Review by rsynek — APPROVED** [review]

**Pull request merged**